### PR TITLE
AKU-963: Added the possibility to show PublishAction only on hover.

### PR DESCRIPTION
**Pull request by dinutasilviu**
Need the fix of https://issues.alfresco.com/jira/browse/AKU-963 issue for rm issue https://issues.alfresco.com/jira/browse/RM-3010

### CODE REVIEW

File: aikau/src/main/resources/alfresco/renderers/PublishAction.js [hunk 1]
```diff
@@ -86,6 +87,29 @@ define(["dojo/_base/declare",
...
+      onlyShowOnHover: false,
+
+      /**
+       * Overrides the default to handle onlyShowOnHover for the current item.
+       *
+       * @instance
+       */
+      postCreate: function alfresco_renderers_PublishAction__postCreate() {
+         this.inherited(arguments);
+         if (this.onlyShowOnHover === true)
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/PublishAction.js]: Don't really need the `=== true` test

File: aikau/src/main/resources/alfresco/renderers/PublishAction.js [hunk 2]
```diff
@@ -86,6 +87,29 @@ define(["dojo/_base/declare",
...
+         {
+            domClass.add(this.domNode, "hover-only");
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/PublishAction.js]: Should be using BEM for this, so the class would be `.alfresco-renderers-PublishAction--show-on-hover` or very similar.

File: aikau/src/main/resources/alfresco/renderers/css/PublishAction.css [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
       vertical-align: text-bottom;
    }
+}
+.alfresco-lists-views-layout-_MultiItemRendererMixin__item {
+   &:hover {
+      .alfresco-renderers-PublishAction.hover-only {
+         > span {
+            visibility: visible;
+         }
+         > img {
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/css/PublishAction.css]: In addition to using BEM (as described above), these two selectors can be combined, i.e. `> span, > img {`

**Comment by MartinDoyleUK** [comment]
It would be good to add tests for this. Have they been attempted?

**Comment by MartinDoyleUK** [comment]
This branch has conflicts on it. Is it possible to please rebase on the latest develop and resolve the conflicts?

**Comment by davidcognite** [comment]
@dinutasilviu - see https://github.com/Alfresco/Aikau/blob/master/aikau/src/test/resources/alfresco/documentlibrary/views/AlfDetailedViewTest.js#L53 for an example of testing hover behaviour.

**Comment by dinutasilviu** [comment]
Thanks Martin and David....will do the changes after I'll finish my current issue.

**Comment by draperd** [comment]
This has been replaced by #1085

**Pull request closed without merging**